packages/backend-data/API.md

```diff
@@ -15,14 +15,25 @@ export type ApiKeyAuthorizationModeProps = {
     expiresInDays?: number;
 };
 
+// @public
+export const AuthorizationModeMapping: {
+    readonly iam: "AWS_IAM";
+    readonly userPool: "AMAZON_COGNITO_USER_POOLS";
+    readonly oidc: "OPENID_CONNECT";
+    readonly apiKey: "API_KEY";
+    readonly lambda: "AWS_LAMBDA";
+};
+
 // @public
 export type AuthorizationModes = {
-    defaultAuthorizationMode?: DefaultAuthorizationMode;
     apiKeyAuthorizationMode?: ApiKeyAuthorizationModeProps;
     lambdaAuthorizationMode?: LambdaAuthorizationModeProps;
     oidcAuthorizationMode?: OIDCAuthorizationModeProps;
     allowListedRoleNames?: string[];
-};
+} & DefineDataAuthConfig;
+
+// @public (undocumented)
+export type CdkDefaultAuthorizationMode = (typeof AuthorizationModeMapping)[DefaultAuthorizationMode];

```

Comment:
we should not use `CDK` in types in this layer.

Reply:
Sure. I had these named `DeprecatedDefaultAuthorizationMode`, which didn't seem right when I was building the translation logic to map these into the names used in the `CDKAuthorizationModes` object. The name of this destination object informed this name. If we are removing support for these capital snake case types, then I should be able to remove this type entirely.  Hopefully this is what we can move to. Will circle back after conferring with Rene.